# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

*****
### What test is appropriate for this problem? Does CLT apply?

In [13]:
print 'Total:{},black sounding names:{},white sounding names:{}'.format(len(data),len(data[data.race=='b']),len(data[data.race=='w']))

Total:4870,black sounding names:2435,white sounding names:2435


The two sample t test should be appropriate for this problem because we have 4870 samples, and each group has more than 30 samples, so is big enough to apply CLT and simulate a normal distribution using t test

*****
### What are the null and alternate hypotheses?

H0: There is no difference in number of request for interviews for black and white job applicants.

H1: There is a significant difference in number of request for interviews for black and white job applicants.

*****
### Compute margin of error, confidence interval, and p-value.

In [29]:
b,w = data[data.race=='b'],data[data.race=='w']
y1,y2 = b.call,w.call
N1,N2 = len(y1),len(y2)
margin_of_error = abs(np.mean(y1)-np.mean(y2))/np.sqrt(np.var(y1)/N1+np.var(y2)/N2)
sp = np.sqrt(((N1-1)*np.var(y1)+(N2-1)*np.var(y2))/float(N1+N2-2))*np.sqrt(1.0/N1+1.0/N2)
CI = [abs(np.mean(y1)-np.mean(y2))-stats.norm.ppf(0.975)*sp,abs(np.mean(y1)-np.mean(y2))+stats.norm.ppf(0.975)*sp]
p_value = stats.norm.pdf(margin_of_error)*2
print CI,p_value

[0.016777728660531926, 0.047287981049589228] 0.00016747621716


*****
### Write a story describing the statistical significance in the context or the original problem.

In [32]:
[CI[i]/np.mean(y1) for i in range(2)]

[0.26021509231785422, 0.73341550595524285]

From two sample t test we can see that with 95% confidence level the difference between white applicants and black applicants range from 0.016to 0.042, which means white applicants can have as high as 73% greater chance than black applicants to get interview requests from employers.

*****
### Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

I think so, because the difference makes up at least 26% of the impact white applicants can have on getting employer's interview requests